apps/dash-yield-curve/app.py

```diff
@@ -0,0 +1,425 @@
+            x=x_secondary,
+            y=y_secondary,
+            z=z_secondary,
+            hoverinfo="x+y+z",
+            line=dict(color="#444444"),
+        )
+
+        data = [trace1, trace2]
+
+    else:
+
+        trace1 = dict(
+            type="contour",
+            x=ylist,
+            y=xlist,
+            z=np.array(zlist).T,
+            colorscale=[
+                [0, "rgb(230,245,254)"],
+                [0.4, "rgb(123,171,203)"],
+                [0.8, "rgb(40,119,174)"],
+                [1, "rgb(37,61,81)"],
+            ],
+            showscale=False,
+            zmax=9.18,
+            zmin=0,
+            line=dict(smoothing=1, color="rgba(40,40,40,0.15)"),
+            contours=dict(coloring="heatmap"),
+        )
+
+        data = [trace1]
+
+    layout = dict(
+        autosize=True,
+        font=dict(size=12, color="#CCCCCC"),
+        margin=dict(t=5, l=5, b=5, r=5),
+        showlegend=False,
+        hovermode="closest",
+        scene=dict(
+            aspectmode="manual",
+            aspectratio=dict(x=2, y=5, z=1.5),
+            camera=dict(up=UPS[value], center=CENTERS[value], eye=EYES[value]),
+            annotations=[
+                dict(
+                    showarrow=False,
+                    y="2015-03-18",
+                    x="1-month",
+                    z=0.046,
+                    text="Point 1",
+                    xanchor="left",
+                    xshift=10,
+                    opacity=0.7,
+                ),
+                dict(
+                    y="2015-03-18",
+                    x="3-month",
+                    z=0.048,
+                    text="Point 2",
+                    textangle=0,
+                    ax=0,
+                    ay=-75,
+                    font=dict(color="black", size=12),
+                    arrowcolor="black",
+                    arrowsize=3,
+                    arrowwidth=1,
+                    arrowhead=1,
+                ),
+            ],
+            xaxis={
+                "showgrid": True,
+                "title": "",
+                "type": "category",
+                "zeroline": False,
+                "categoryorder": "array",
+                "categoryarray": list(reversed(xlist)),
+            },
+            yaxis={"showgrid": True, "title": "", "type": "date", "zeroline": False},
+        ),
+    )
+
+    figure = dict(data=data, layout=layout)
+    return figure
+
+
+# Make annotations
+@app.callback(Output("text", "children"), [Input("slider", "value")])
+def make_text(value):
+    if value is None:
+        value = 0
+
+    return TEXTS[value]
+
+
+# Button controls
+@app.callback(
+    Output("slider", "value"),
+    [Input("back", "n_clicks"), Input("next", "n_clicks")],
+    [State("slider", "value")],
+)
+def advance_slider(back, nxt, slider):
+
+    if back is None:
+        back = 0
+    if nxt is None:
+        nxt = 0
+    if slider is None:
+        slider = 0
+
+    global last_back
```

Comment:
Please use `dcc.Store` for these instead!

Reply:
This is leading to some odd behaviour; when you click on another point in the slider, then click back to the first point, then hit next, nothing happens.